tests/iter.rs

```diff
@@ -0,0 +1,42 @@
+use std::{
+    future::Future,
+    pin::{pin, Pin},
+    task::{Context, Poll},
+};
+
+use futures_util::stream;
+use streamunordered::StreamUnordered;
+
+// An `async move { i }` causes unpin issues with `iter_mut` and `iter_mut_with_token`.
+pub struct UnpinFuture(usize);
+
+impl Future for UnpinFuture {
+    type Output = usize;
+
+    fn poll(self: Pin<&mut Self>, _: &mut Context<'_>) -> Poll<Self::Output> {
+        Poll::Ready(self.0)
+    }
+}
+
+#[test]
+fn iter_methods() {
+    let mut streams = StreamUnordered::new();
+    for i in 0..5 {
+        streams.insert(stream::once(UnpinFuture(i)));
+    }
+
+    let result: (Vec<_>, Vec<_>) = streams.iter_with_token().unzip();
+    assert_eq!(result.1, vec![5, 4, 3, 2, 1]);
```

Comment:
I don't think we want to guarantee anything about the _order_ in which `iter_with_token` yields the streams. Just call `sort` on the `Vec` before you compare it and you should be fine :+1:

Reply:
Same for the tests below.